lib/sprockets/sass/registration.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+module Sprockets
+  module Sass
+    # class useful for registering engines, tranformers, preprocessors and conpressors for sprockets
+    # depending on the version of sprockets
+    class Registration
+      attr_reader :klass
+
+      AVAILABLE_VERSIONS = [4, 3, 2].freeze
+
+      DEFAULT_ACTION = { min_comparison: '>=', min: 2, max_comparison: '<', max: 3, action: :register_sprockets_legacy }.freeze
+
+      ACTIONS = [
+        { min_comparison: '>=', min: 3, max_comparison: '<', max: 4, action: :register_sprockets_v3 },
+        #    { min_comparison: '>=' , min: 4,  action: :register_sprockets_v4 }
+      ].freeze
+
+      attr_reader :klass, :sprockets_version, :registration_instance, :action_details, :version_selected
+
+      def initialize(klass)
+        @klass = klass
+        @sprockets_version = Sprockets::Sass::Utils.version_of_sprockets
+        @registration_instance = self
+        @action_details = action_to_be_executed
+        @version_selected = @action_details[:min]
+      end
+
+      def run
+        require_libraries
+        send(@action_details[:action])
+      end
+
+      def require_libraries
+        AVAILABLE_VERSIONS.each do |version|
+          require_standard_libraries(version)
+        end
+        require 'sprockets/sass/functions'
+      end
+
+    private
+
+      def require_standard_libraries(version)
+        %w(cache_store compressor functions importer sass_template scss_template).each do |filename|
+          begin
+            require "sprockets/sass/v#{version}/#{filename}"
+          rescue LoadError; end
+        end
+      end
+
+      def register_sprockets_v3_common
+        %w(sass scss).each do |mime|
+          _register_mime_types(mime_type:  "application/#{mime}+ruby", extensions: [".#{mime}.erb", ".css.#{mime}.erb"])
+        end
+        _register_compressors(mime_type: 'text/css', name: :sprockets_sass, klass: Sprockets::Sass::Utils.get_class_by_version('Compressor'))
+      end
+
+      def register_sprockets_v4
+        register_sprockets_v3_common
+        %w(sass scss).each do |mime|
+          _register_transformers(from: "application/#{mime}+ruby", to: "text/#{mime}", klass: Sprockets::ERBProcessor)
+        end
+        _register_v4_preprocessors(
+          Sprockets::Sass::Utils.get_class_by_version('SassTemplate') => ['text/sass', 'text/css'],
```

Comment:
I could , but the reason why it is like that, is that if a greater version will be released, if nothing changes for that version, this will use the last version that is higher.   I can do that though if you want . Just that you'll have to change that again if you will need to use another version. 

So this will automatically pick the higher version available. But if you think hardcoding it is better, i am fine with that.